litho-core/src/main/java/com/facebook/litho/ComponentTree.java

```diff
@@ -1923,11 +1923,11 @@ protected LayoutState calculateLayoutState(
               extraAttribution);
 
       synchronized (mLayoutStateFutureLock) {
-        if (localLayoutStateFuture.equals(mLayoutStateFuture)) {
-          // Use the latest LayoutState calculation if it's the same.
+        if (localLayoutStateFuture.equals(mLayoutStateFuture) && !mLayoutStateFuture.isReleased()) {
```

Comment:
Just to be sure, this is best effort only -- we could release before calling run/get below

Reply:
Yep, that's why there's still the null check on `layoutState` afterwards.